compiler/src/main/java/com/squareup/anvil/compiler/ModuleMergerIr.kt

```diff
@@ -202,7 +214,7 @@ internal class ModuleMergerIr(
         )
         .flatMap { classSymbol ->
           val annotation = classSymbol.owner.annotation(annotationFqName)
-          if (scope == annotation.scope()) {
+          if (scope.fqName == annotation.scope()) {
```

Comment:
You should compare `ClassReferenceIr` instances and not FqName.

Reply:
This should be updated in #544. I just updated usages for `fqName` here to reduce the scope of this PR :).

Reply:
Sounds good 👍🏻